linode/nbnode/framework_resource.go

```diff
@@ -0,0 +1,260 @@
+package nbnode
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: helper.NewBaseResource(
+			helper.BaseResourceConfig{
+				Name:   "linode_nodebalancer_node",
+				IDType: types.StringType,
+				Schema: &frameworkResourceSchema,
+			},
+		),
+	}
+}
+
+type Resource struct {
+	helper.BaseResource
+}
+
+func AddNodeResource(ctx context.Context, node linodego.NodeBalancerNode, resp *resource.CreateResponse, plan ResourceModel) {
+	resp.State.SetAttribute(ctx, path.Root("id"), types.StringValue(strconv.Itoa(node.ID)))
+}
+
+func (r *Resource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	tflog.Debug(ctx, "Create "+r.Config.Name)
+
+	var plan ResourceModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	nodeBalancerID, configID, createOpts := plan.GetCreateParameters(&resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, "client.CreateNodeBalancerNode(...)", map[string]any{
+		"options": createOpts,
+	})
+	node, err := client.CreateNodeBalancerNode(ctx, nodeBalancerID, configID, createOpts)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to Create a Linode NodeBalancerNode", err.Error())
+		return
+	}
+
+	// Add resource to TF states earlier to prevent
+	// dangling resources (resources created but not managed by TF)
+	AddNodeResource(ctx, *node, resp, plan)
```

Comment:
Just curious, is preventing dangling resources in creation a new suggestion from Hashicorp? Looks like we didn't have this logic in other resources.

Reply:
Hi Ye, that's not a official suggestion, but I feel it would be safer to have the ID set into state as soon as they are available to prevent any possible errors returned before reaching `resp.State.Set(ctx, &plan)`.

I wrote a similar pattern in the disk resource: https://github.com/linode/terraform-provider-linode/blob/58fc6dee71f2d7a57712bf5f801866cf13e018af/linode/instancedisk/framework_resource.go#L124